pkg/cvo/cvo.go

```diff
@@ -499,28 +499,54 @@ func (optr *Operator) clusterOperatorEventHandler() cache.ResourceEventHandler {
 	return cache.ResourceEventHandlerFuncs{
 		UpdateFunc: func(old, new interface{}) {
 			versionName := "operator"
-			_, oldVersion := clusterOperatorInterfaceVersionOrDie(old, versionName)
-			newStruct, newVersion := clusterOperatorInterfaceVersionOrDie(new, versionName)
+			oldStruct := clusterOperatorInterfaceStructOrDie(old)
+			newStruct := clusterOperatorInterfaceStructOrDie(new)
+
+			oldVersion := clusterOperatorVersion(oldStruct, versionName)
+			newVersion := clusterOperatorVersion(newStruct, versionName)
 			if optr.configSync != nil && oldVersion != newVersion {
 				msg := fmt.Sprintf("Cluster operator %s changed versions[name=%q] from %q to %q", newStruct.ObjectMeta.Name, versionName, oldVersion, newVersion)
 				optr.configSync.NotifyAboutManagedResourceActivity(new, msg)
+				return
+			}
+
+			oldAvailable := clusterOperatorConditionStatus(oldStruct, configv1.OperatorAvailable)
+			newAvailable := clusterOperatorConditionStatus(newStruct, configv1.OperatorAvailable)
+			if optr.configSync != nil && oldAvailable != newAvailable {
```

Comment:
We might want to do this same sort of thing for `Degraded` too while we're at it.